finesse/gui/hardware_set/hardware_set.py

```diff
@@ -82,6 +82,7 @@ def _non_empty(x: Any) -> bool:
 
 _hw_set_schema = Schema(
     {
+        "version": int,
```

Comment:
Let's enforce the schema version. Users may attempt to load a file made with a newer version of FINESSE, with unexpected results.

I *think* you can just write this:

```suggestion
        "version": 1,
```

(Though it may be better to use a constant for the magic number.)

See: https://github.com/keleshev/schema?tab=readme-ov-file